libs/stringtable/src/rapify.rs

```diff
@@ -0,0 +1,161 @@
+use crate::{Key, Project};
+use hemtt_workspace::WorkspacePath;
+use tracing::warn;
+
+static ALL_LANGUAGES: [&str; 14] = ["English", "Czech", "French", "Spanish", "Italian", "Polish", "Portuguese", "Russian", "German", "Korean", "Japanese", "Chinese", "Chinesesimp", "Turkish"];
+
+#[derive(Default)]
+struct XmlbLayout {
+    //  4 byte numbers, little end?, nul term strings
+    header: Vec<u8>,            // Version
+    languages: Vec<u8>,         // Language Count, [Languages]
+    offsets: Vec<u8>,           // Offset Count, [Offsets]
+    keys: Vec<u8>,              // Key Count, [Keys]
+    translations: Vec<Vec<u8>>, // [Translation Count, [Translations], ...]
+}
+
+/// # Panics
+pub fn convert_stringtable(project: &Project, xml_path: &WorkspacePath) {
+    let result = rapify(project);
+
+    if result.is_some() {
+        // Create stringtable.bin
+        let xmlb_path = xml_path.with_extension("bin").expect("vfs error");
+        let mut xmlb_file = xmlb_path.create_file().expect("vfs error");
+
+        // Remove Original stringtable.xml
+        xml_path.vfs().remove_file().expect("vfs error");
+
+        let data = result.expect("data struct valid");
+        xmlb_file.write_all(&data.header).expect("IO Error");
+        xmlb_file.write_all(&data.languages).expect("IO Error");
+        xmlb_file.write_all(&data.offsets).expect("IO Error");
+        xmlb_file.write_all(&data.keys).expect("IO Error");
+        for translation_buffer in &data.translations {
+            xmlb_file.write_all(translation_buffer).expect("IO Error");
+        }
+        println!("pass");
+    } else {
+        warn!("fail");
+    }
+}
+
+/// Write string with null-termination
+fn write_string(buffer: &mut Vec<u8>, input: &str) {
+    buffer.extend(input.as_bytes());
+    buffer.push(0);
+}
+fn write_int(buffer: &mut Vec<u8>, input: i32) {
+    buffer.extend(&input.to_le_bytes());
+}
+
+fn rapify(project: &Project) -> Option<XmlbLayout> {
+    let mut data: XmlbLayout = XmlbLayout::default();
+
+    // Restructure translations: flat for each language
+    let mut all_keys: Vec<String> = Vec::new();
```

Comment:
Could be nice to preallocate some space here, but the speed up will be pretty small